tests/test_datasets/test_dataset.py

```diff
@@ -1684,3 +1687,193 @@ def test_check_timestamp_type_warning():
+- ts.raw_df получается путем alignment-а
+- ts.df_exog -- коллизия имени с original_timestamp_name
+"""
+
+
+@pytest.mark.parametrize(
+    "df_name, freq, original_timestamp_name, future_steps",
+    [
+        ("df_aligned_datetime", "D", "external_timestamp", 1),
+        ("df_aligned_int", None, "external_timestamp", 1),
+        ("df_misaligned_datetime", "D", "external_timestamp", 1),
+        ("df_misaligned_int", None, "external_timestamp", 1),
+        ("df_misaligned_datetime", "D", "new_timestamp", 1),
+        ("df_misaligned_datetime", "D", "external_timestamp", 3),
+        ("df_misaligned_datetime", "D", "external_timestamp", 100),
+    ],
+)
+def test_create_from_misaligned_without_exog(df_name, freq, original_timestamp_name, future_steps, request):
+    df = request.getfixturevalue(df_name)
+    ts = TSDataset.create_from_misaligned(
+        df=df, df_exog=None, freq=freq, original_timestamp_name=original_timestamp_name, future_steps=future_steps
+    )
+
+    alignment = infer_alignment(df)
+    expected_raw_df = TSDataset.to_dataset(apply_alignment(df=df, alignment=alignment))
+    pd.testing.assert_frame_equal(ts.raw_df, expected_raw_df)
+
+    timestamp_df = make_timestamp_df_from_alignment(
+        alignment=alignment,
+        start=expected_raw_df.index[0],
+        periods=len(expected_raw_df) + future_steps,
+        freq=freq,
+        timestamp_name=original_timestamp_name,
+    )
+    expected_df_exog = TSDataset.to_dataset(timestamp_df)
+    pd.testing.assert_frame_equal(ts.df_exog, expected_df_exog)
+
+    assert original_timestamp_name in ts.known_future
+    assert ts.freq is None
+
+
+@pytest.mark.parametrize(
+    "df_name, df_exog_name, freq, known_future, original_timestamp_name, future_steps",
+    [
+        ("df_aligned_datetime", "df_exog_aligned_datetime", "D", ["exog_1"], "external_timestamp", 1),
+        ("df_aligned_datetime", "df_exog_misaligned_datetime", "D", ["exog_1"], "external_timestamp", 1),
+        ("df_aligned_int", "df_exog_aligned_int", None, ["exog_1"], "external_timestamp", 1),
+        ("df_aligned_int", "df_exog_misaligned_int", None, ["exog_1"], "external_timestamp", 1),
+        ("df_misaligned_datetime", "df_exog_aligned_datetime", "D", ["exog_1"], "external_timestamp", 1),
+        ("df_misaligned_datetime", "df_exog_misaligned_datetime", "D", ["exog_1"], "external_timestamp", 1),
+        ("df_misaligned_int", "df_exog_aligned_int", None, ["exog_1"], "external_timestamp", 1),
+        ("df_misaligned_int", "df_exog_misaligned_int", None, ["exog_1"], "external_timestamp", 1),
+        ("df_misaligned_datetime", "df_exog_misaligned_datetime", "D", ["exog_1"], "new_timestamp", 1),
+        ("df_misaligned_datetime", "df_exog_misaligned_datetime", "D", ["exog_1"], "external_timestamp", 3),
+        ("df_misaligned_datetime", "df_exog_misaligned_datetime", "D", ["exog_1"], "external_timestamp", 100),
+    ],
+)
+def test_create_from_misaligned_with_exog(
```

Comment:
Is it fine, that we check `df_exog` and `raw_df` in the same test?

Reply:
We check all parameters in the same test. I doesn't see any benefit in separating it right now.

Reply:
You're right, if we split it into two separate tests, then we get duplication of code.